src/Infrastructure/Jobs/NotifyAccountDeactivationJob.cs

```diff
@@ -18,34 +17,41 @@ public async Task Execute(IJobExecutionContext context)
 
         if (context.RefireCount > 3)
         {
-            logger.LogWarning($"Failed to complete within 3 tries, aborting...");
+            logger.LogWarning($"Failed to complete notifying accounts that are due to deactivate within 3 tries, aborting...");
             return;
         }
 
         try
         {
-            logger.LogInformation("Notifying accounts that will be deactivated soon");
+            logger.LogInformation("Starting notifying accounts that will be deactivated soon");
 
-            DateTime sevenDaysFromDeactivationDate = DateTime.Now
-                .AddDays(-30) // 30 days ago
-                .AddDays(7); // 7 days before
+            DateTime sevenDaysFromDeactivationDate = DateTime.Today.AddDays(-23); // exactly 23 days ago, ignoring time
 
             var users = await userManager.Users
                 .IgnoreAutoIncludes()
                 .Where(user => user.IsActive)
-                .Where(user => user.LastLogin < sevenDaysFromDeactivationDate
-                    || (user.LastLogin == null && user.Created < sevenDaysFromDeactivationDate))
+                .Where(user =>
+                    (user.LastLogin.HasValue && user.LastLogin.Value.Date == sevenDaysFromDeactivationDate) ||
+                    (!user.LastLogin.HasValue && user.Created.HasValue && user.Created.Value.Date == sevenDaysFromDeactivationDate))
```

Comment:
```suggestion
                .Where(user =>
                    user.LastLogin.HasValue
                        ? user.LastLogin.Value.Date == sevenDaysFromDeactivationDate
                        : user.Created.HasValue && user.Created.Value.Date == sevenDaysFromDeactivationDate)
```

Improve readability of where clause